src/jit/emitxarch.cpp

```diff
@@ -4797,6 +4937,16 @@ void emitter::emitIns_AX_R(instruction ins, emitAttr attr, regNumber ireg, regNu
 }
 
 #if FEATURE_HW_INTRINSICS
+void emitter::emitIns_SIMD_R_R(instruction ins, regNumber reg1, regNumber reg2, var_types simdtype)
+{
+    emitIns_R_R(ins, emitTypeSize(simdtype), reg1, reg2);
+}
+
+void emitter::emitIns_SIMD_R_R_I(instruction ins, regNumber reg1, regNumber reg2, ssize_t iVal, var_types simdtype)
+{
+    emitIns_R_R_I(ins, emitTypeSize(simdtype), reg1, reg2, (int)iVal);
```

Comment:
I believe `R_R_I` should only happen under `UseVEXEncoding`

Reply:
Depends on instruction - all shuffle cases, `cmppd`, `cmpps`, `pinsrw` are `IF_RRW_RRW_CNS` so they should go through `R_R_I`

Reply:
Yes, but those instructions should be going through `SIMD_R_R_R_I`.

In almost all cases so far, the codegen has been calling the `emitIns_SIMD_*` case that matches the VEX encoding. We then have logic for the fallback case so that args can be arranged properly.

Ex: `cmppd` is `R_R_R_I` on AVX+ machines and `R_R_I` elsewhere. So we call `emitIns_SIMD_R_R_R_I`.

`emitIns_SIMD_R_R_R_I` then has the logic for, on vex, generate the instruction directly (emtIns_R_R_R_I), on non-vex generate a `movaps` (emitIns_R_R) followed by the instruction (emitIns_R_R_I).

That being said, it does look like the `pshufd` instructions are bit special in that they (like crc32, lzcnt, popcnt, and a few others) have a consistent encoding between VEX and non-VEX machines, in which case we should just be calling `emitIns_*` directly, without an intermediary `emitIns_SIMD_*` step